### PR TITLE
refactor(cli)!: collapse the executor mode ladder; retire the interactive handoff

### PR DESCRIPTION
**Pull request by advl**
## Done

Final PR of the P.05 train. With byte-equality landed (#761), this retires the last duplicated prompting surface so there is exactly one execution core and one prompting model.

- Retire the `InteractiveSpec` / `createInteractiveResult` / `InteractiveHandler` handoff entirely. `CommandResult` collapses from three variants to two (`output | exit`).
- Collapse `executeGenerator`'s mode ladder to **preview modes (llm / json / dry-run) × execution**. Interactive prompting now flows through a caller-injected `PromptSession` factory on `CommandContext.promptSession`; cli-core defines the contract and stays UI-free, while pragma injects its readline `createInteractivePromptSession` in `runCli`.
- Interactivity gate (per [P.05](https://github.com/advl/pragma-adrs/blob/main/session/P/P.05.UNIFIED_PROMPTING_WIZARD.md)): the wizard engages only when **stdin and stdout are both TTYs**. Piped stdin takes the non-interactive path; a missing required flag exits **3** naming the exact flags to pass. Ctrl-C exits **130**.
- Delete pragma's bespoke `runInteractiveCommand` (+ pipeline plumbing) and `createInteractiveResult`; `registerAll.handleResult` reduces to `output | exit`.
- **summon is untouched** — it renders its own Ink `<App>` and never used the handoff.

Net −1,021 lines.

Drive-bys (from the adversarial review of this diff):
- Add a regression test asserting `handleResult` propagates a non-zero exit result to `process.exitCode` — the sole path by which exit 3/130 reach the process status, previously exercised only with code 0.
- Fix cli-core's README, which still described `CommandResult` as a three-variant union listing the retired `interactive` variant.

Fixes the last item of the P.05 prompting-unification plan.

## QA

- `pragma create component react src/components/Button --dry-run` → preview, exit 0.
- Interactive wizard in a **real terminal**: `pragma create package` renders text/select/confirm prompts on stderr; typed answers are honored; Ctrl-C aborts with exit **130**.
- `pragma create package < /dev/null` → completes via the non-interactive path (no hang, no half-wizard).
- Exit **3** on a missing required flag applies to a required prompt **without** a default (the shipped component/package generators default every prompt, so use a generator with a no-default required prompt to observe it; otherwise covered by `executeGenerator.test.ts`).
- Automated: `cd packages/cli/core && bun run test` (265 tests, 100% coverage); `cd packages/cli/pragma && bun run test` (goldens + cross-binary byte-equality green).

### PR readiness check

- [x] PR should have one of the following labels: `Breaking Change 💣`.
- [x] PR title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) format.
- [x] The code follows the appropriate [code standards](https://github.com/canonical/web-code-standards).
- [x] All packages define the required scripts in `package.json` (`check`, `check:fix`, `test`; build packages: `build`, `build:all`).
- [x] No new package introduced.
- [x] No visual testing required — `no visual change` label added to skip Chromatic.

## Screenshots

N/A — executor-internal refactor, no visual change.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01DF9ExVCukzqpe1Fus9V1no

---
_Generated by [Claude Code](https://claude.ai/code/session_01DF9ExVCukzqpe1Fus9V1no)_

### CODE REVIEW

File: packages/cli/core/src/lib/executeGenerator.ts [hunk 1]
```diff
@@ -172,6 +181,93 @@ const suppressTaskLogs = (): void => {
...
+  params: Record<string, unknown>,
+  ctx: CommandContext,
+): Promise<CommandResult> => {
+  const session = ctx.promptSession?.();
+  if (session === undefined || !isInteractiveTerminal()) {
+    process.stderr.write(
+      `${formatInteractiveUnavailable(gen.prompts, cliAnswers)}\n`,
+    );
+    return createExitResult(3);
+  }
```
[the remark this answers is not in the document]
Comment (advl) [reply packages/cli/core/src/lib/executeGenerator.ts]: Good catch on both — fixed in a1475520.

- **Eager construction:** `runInteractiveExecution` now computes `onTty = isInteractiveTerminal()` first and only calls `ctx.promptSession?.()` when on a TTY. Off a TTY the factory is never invoked, so no readline handle is opened and discarded undisposed. The test that previously froze this behavior (asserting `dispose` not called after construction) now asserts the factory is never called at all.
- **Misleading header:** the two failure modes get distinct headers — `"Interactive mode is not available on a non-interactive terminal."` for a non-TTY, and `"No interactive prompt session is available."` for a TTY with no injected session. Both still list the missing required flags and exit 3.

cli-core stays at 265 tests / 100% coverage.

— Claude Code

---
_Generated by [Claude Code](https://claude.ai/code)_
